webapp/channels/src/utils/message_html_to_component.tsx

```diff
@@ -29,25 +57,25 @@ import PostEditedIndicator from 'components/post_view/post_edited_indicator';
  * - hasPluginTooltips - If specified, the LinkTooltip component is placed inside links. Defaults to false.
  * - channelId = If specified, to be passed along to ProfilePopover via AtMention
  */
-export function messageHtmlToComponent(html, options = {}) {
+export function messageHtmlToComponent(html: any, options: Options = {}) {
     if (!html) {
         return null;
     }
 
-    const parser = new Parser();
-    const processNodeDefinitions = new ProcessNodeDefinitions(React);
+    const parser = new (Parser as any)();
+    const processNodeDefinitions = new (ProcessNodeDefinitions as any)();
 
     function isValidNode() {
         return true;
     }
 
-    const processingInstructions = [
+    const processingInstructions: ProcessingInstruction[] = [
 
         // Workaround to fix MM-14931
         {
             replaceChildren: false,
-            shouldProcessNode: (node) => node.type === 'tag' && node.name === 'input' && node.attribs.type === 'checkbox',
-            processNode: (node) => {
+            shouldProcessNode: (node: any) => node.type === 'tag' && node.name === 'input' && node.attribs.type === 'checkbox',
+            processNode: (node: any) => {
```

Comment:
I don't like the use of any here, but looking at the type definitions on the original library, seems like we cannot do much better. 0/5

Reply:
same

Reply:
Yes, that is the problem, the original library.